extensions/amp-image-slider/0.1/test/integration/test-amp-image-slider.js

```diff
@@ -901,23 +899,20 @@ config.run('amp-image-slider', function() {
     // Rejects if timeout
     function startObserver(
       target, cb, opt_errorMessage) {
-      cleanupObserver();
-
-      const deferred = new Deferred();
-      observer = new win.MutationObserver(mutationList => {
-        if (cb(mutationList)) {
+      return new Promise((resolve, reject) => {
+        observer = new win.MutationObserver(mutationList => {
+          if (cb(mutationList)) {
+            cleanupObserver();
+            resolve();
+          }
+        });
+        observerTimeout = win.setTimeout(() => {
+          // Cancel observer when times out
           cleanupObserver();
-          deferred.resolve();
-        }
+          reject(new Error(opt_errorMessage || 'Observer times out'));
+        }, 1600);
```

Comment:
Nit: this can be a `const` on top of the file, so we're consistent about this timeout.